### ISSUE TITLE
Callback called twice in rule evaluation

### ISSUE THREAD

**Issue by anandrajneesh** [anandrajneesh/decision-engine#8]
```javascript
RuleEngine.run = function (fact, group, cb) {
    async.concat(
            _rules[group],
            function (rule, cb) {
              rule.parsedCondition.evaluate(fact, function (err, result) {
                if (result) cb(undefined, rule.name)
                if (err) console.log(err)
                else cb()
              })
            },
            cb)
  }
```
The else part is a bug. If result is present callback is called, then code continues on to check for err which on failing executes the else part hence calling the callback again.

**Comment by anandrajneesh** [comment]
Current test cases do not capture this error, need to add reproducible test cases.

**Comment by anandrajneesh** [comment]
Added a delay in done() invocation in one of the test cases to let async fail if callback is called twice.
Fixed in v1.1.4
